src/lib/components/Menu/Menu.js

```diff
@@ -77,129 +78,350 @@ const mapStateToProps = (state, ownProps) => {
+   */
+  insertGroupCount(layer, groupCounter) {
+    Object.keys(layer).forEach(key => {
+      layer[key].count = groupCounter;
+      groupCounter += 1;
+
+      layer[key].layers.forEach(keyLayer => {
+        if (!keyLayer.id) {
+          keyLayer = this.insertGroupCount(keyLayer, groupCounter);
+        }
+      });
+    });
+
+    return {
+      layer,
+      groupCounter,
+    };
+  }
+
+  /**
+   * Loop over the categories and modify data
+   * @param {array} categories Menu categories and their layers, basically all menu items
+   * @returns {array} Modified categories
+   */
+  parseCategories(categories) {
+    let groupCounter = 0;
+    categories.forEach(category => {
+      category.layers.forEach(layer => {
+        if (!layer.id) {
+          // This is a group. Give it and its nested groups a count
+          // This will make sure when we toggle a group, another group with the
+          // exact same name is not toggled. The array of open groups is stored as an
+          // array in the store. Another implementaion would to give each group an Id e.g a random no but
+          // this means different menu instances will have groups that have different Ids which we do not
+          // want (We would like menu instances under the same map ID to behave the same. If say I open a group in one
+          // menu, it should appear open the other ). A contigous int counter is
+          // therefore used to ensure the counter for a group in different menu
+          // instances is the same, and also, groups in the same menu instances do not share a count.
+          const obj = this.insertGroupCount(layer, groupCounter);
+          layer = obj.layer;
+          groupCounter = obj.groupCounter;
+        }
+      });
+    });
+
+    return categories;
+  }
+
+  /**
+   * Check if a user has permission to access layer
+   * @param {*} authConfigs - Authentication configurations
+   * @param {*} userInfo - User details
+   */
+  canAccessLayer(layer, authConfigs, userInfo) {
+    const users = authConfigs.LAYERS[layer.id];
+
+    return (
+      (users && userInfo && users.includes(userInfo.username)) ||
+      (authConfigs.LAYERS &&
+        authConfigs.LAYERS.ALL &&
+        authConfigs.LAYERS.ALL.includes(userInfo.username))
+    );
+  }
+
+  /**
+   * Return an accesible group layer. If the layer has no accessible children
+   * return false, else return the modified layer with the accessible children
+   * @param {*} layer - Group layer
+   * @param {*} authConfigs - Authentication configaurations
+   * @param {*} userInfo - Auth user details
+   */
+  getAccessibleGroupLayer(layer, authConfigs, userInfo) {
+    const keys = Object.keys(layer);
+    const accessibleKeys = [];
+
+    keys.forEach(key => {
+      const accessibleKeySubLayers = [];
+
+      layer[key].layers.forEach(subLayer => {
+        if (!subLayer.id) {
+          let groupLayer = this.getAccessibleGroupLayer(subLayer, authConfigs, userInfo);
```

Comment:
`let` => `const` b/c it's not being redeclared in this block. Shouldn't the linter catch this? 🤔